lib/carin_for_blue_button_test_kit/carin_search_test.rb

```diff
@@ -164,57 +164,107 @@ def run_include_search(search_params, param_value)
       assert_response_status(200)
       assert_resource_type(:bundle)
 
-      returned_resources = extract_resources_from_bundle(bundle: resource, response:).select do |item|
-        item.resourceType == resource_type
-      end
+      returned_resources_all = extract_resources_from_bundle(bundle: resource, response: response)
+      returned_resources_resource_type = returned_resources_all.select{|item| item.resourceType == resource_type}
 
-      skip_if returned_resources.blank?, no_resources_message
+      skip_if returned_resources_resource_type.blank?, self.no_resources_message
 
-      returned_resources.each do |resource|
+      returned_resources_resource_type.each do |resource|
         match_found = false
+        reference_bool = true
         paths = include_param_paths(param_value)
-
+        
         if param_value != 'ExplanationOfBenefit:*'
           paths.each do |path|
-            values_found = resolve_path(resource.source_hash, path)
+              values_found = resolve_path(resource.source_hash, path)
 
-            match_found = values_found.length.positive?
+              values_found.each do |reference|
```

Comment:
line 181 - 188
```ruby
values_found.each do |reference|
   reference_found = find_included_resource(reference, returned_resources_all)

    # If at least one reference is not found, set reference_bool to false and do not change back to true for any other found references
     if !reference_found
        reference_bool = false
      end
end
```
could be simplified to use `all?` method
```ruby
reference_bool = values_found.all? { |reference| find_included_resource(reference, returned_resources_all) }
```

The potential return values for the `find_included_resource` method are:

- An assertion error (if either of the assertions fails). In this case the test execution will be interrupted. 
- true (if a matching reference is found).
- false (if no matching reference is found and no assertions are raised).

Using `all?`, `reference_bool` will be set to `true ` only if `find_included_resource` returns `true `for all elements in `values_found`. If `find_included_resource` returns `false` for any element, then `reference_bool` will be set to false.